### PR TITLE
Bc hotfix lfs fix

### PR DESCRIPTION
**Pull request by nathan-miller23**
# BC refactor
several structure adjustments and code improvements for the data-oriented portion of the BC pipeline

## Refactor
Some changes to the repo structure:
* the `human_aware_rl/data` directory has been entirely deleted. Human data has been migrated, outdated BC runs have been removed outright as they were already deprecated (still located in Neurips2019 branch)
* All human data is now located in the `human_aware_rl/static/human_data` directory. This is subdivided into `raw` (CSV) and `cleaned` (pickled DataFrame) subdirectories, as well as a smaller `dummy` folder for faster, for efficient tests
* Global variables and hardcoded constants (i.e. `BC_PARAMS`) have been updated to reflect the new paths
* `process_human_trials.py` (code for imputing INTERACTs to start cooking) has been migrated to `human_aware_rl/human/` directory to match all other python data-oriented utils
* All pickle files are stored directly in the repo, as well as the dummy CSVs
* The full, raw CSVs are stored using git Large File System

## New layout data incorporated
Both the 2019 neurips data and the 2020 experimental data now have first class status in this repo. The naming scheme is simple: `2019_hh*` refers to the former, `2020_hh*` the latter

## Forward compatibility
The 2019 data was updated so that its schema matches that of the 2020 data exactly. The script to perform this migration was cleaned up and exposed publicly in `human_aware_rl/human/human_data_forward_compat.py`

## New Functionality
Some additional functionality was added to the data processing utils, primarily in `human_aware_rl/human/processes_dataframs.py`. Most notably:
* `csv_to_df_pickle` 
  * High level function gives first-class public support for working with CSV data directly
  * Performs loading of CSV, pre-processing, filtering, and saving of pickled pandas dataframes all in one call
* `format_trials_df`
  * More streamlined version of `format_hh_trials_df`
  * No longer separate functions for the hh and hai cases
  * Adds additional computed columns involving interactivity (such as `timsteps_since_interact` and `buttons_pressed_per_timestep`)
* `filter_trials/filter_transitions`
  * Top level function for filtering dataframe
  * Accepts arbitrary `filter` function
  * Can either filter by-trial or by-timestep
 
## Docs
Separate docs were added at `human_aware_rl/static/human_data/README.md` detailing the shcema and pre-processing tools available

An example Jupyter Notebook demo-ing the data processing and exploration workflow was included in `human_aware_rl/human/data_wrangling.ipynb`

## Tests
An entire human-data testing suite was written in `human_aware_rl/human/tests.py` and added to the CI workflow

## Outstanding TODOs
* Verify with @mesutyang97 that the functionality of `process_human_trials.py` was not compromised by the migration
* Finalize the 'utils' portion of the human data README
* Investigate why the github workflows integration tests are failing. They're all passing locally, so it appears its an issue setting up the virtual environment data files

### CODE REVIEW

**Comment by nathan-miller23** [comment]
LInking corresponding Overcooked-ai [PR](https://github.com/HumanCompatibleAI/overcooked_ai/pull/76) that is necessary for data processing on new layouts with tomatoes

File: human_aware_rl/human/process_human_trials.py
Comment (nathan-miller23) [inline human_aware_rl/human/process_human_trials.py:1]: @mesutyang97 I converted you file into an argparse script and added a corresponding unit test. However, the thing breaks on the dummy dataset I included for testing purposes (its a random subsample of the 2019 neurips pickle data). 

Unfortunately, on github, all the data-oriented tests are throwing a key error, which tells me there's something wrong with the CI checkout process (maybe because the pickle files are almost near the max github file size). Therefore, you won't be able to see the error I'm referring to in the integration check. I've thus included a screenshot of the error from my local tests. I doubt the screenshot will be super enlightening, so hopefully we can figure out both the github CI issues and then more efficiently tackle this bug


Screen Shot 2021-05-25 at 3 30 34 PM

**Comment by nathan-miller23** [comment]
## Changes PR opened
* Finished the outstanding portion of the human data docs
* Fixed bugs related to imports and naming issues from refactor
* Fixed bug related to tomato featurization (requires corresponding changes in Overcooked-ai code to work here)
* All (but one) test pass locally when proper data files are loaded and proper overcooked-ai commit is checked out

## Oustanding TODOs:
* Why the continued issues with loading the data on the CI checks?
* Fix bug in mesut's insert interact script

File: human_aware_rl/human/process_human_trials.py
Comment (mesutyang97) [reply human_aware_rl/human/process_human_trials.py:1]: I think this is because when I make the conversion, if player doesn't hold an object, the PlayerState simply won't have the "held_object" key. But if we want to stay consistent and always have that key (and just have the value be None when the player isn't holding an object), we can simply add a checking condition in the process_held_object function. 

Let me know what's the desired behavior, and I will update this file accordingly

**Comment by nathan-miller23** [comment]
# Additional Changes
* Fixed bug with git LFS integrating with github CI
* Updated all 2019 CSV data so that all OvercookedStates are encoded using the modern schema
* Upgraded data utils to be compatible with refactored trajectory utils in overcooked-ai 
* Added `verbose` flags throughout this repo and overcooked-ai to optionally silence logging for cleaner test output
* Fixed bugs in `process_human_trials` script (the one that adds additional 'interact' actions)

# Outstanding TODOs
None! All tests are passing, should be ready to merge once reviewed and the corresponding `overcooked-ai` PR is merged

File: human_aware_rl/human/process_human_trials.py
Comment (nathan-miller23) [reply human_aware_rl/human/process_human_trials.py:1]: Yeah not sure if you saw my later follow-ups, but the issue was a schema mismatch. The old schema would simply not have a `held_objects` key if the player wasn't holding anything, the new schema has `held_object : None` in that case. 

I ported __all__ of the data in the repo to the new schema, and, just to be safe for legacy reasons, updated your script so that it can handle both legacy and current schema, depending on whether the `forward-port` flag is passed in or not

**Review by micahcarroll — APPROVED** [review]
This looks good to me! 👍 I like the new get_bc_params signature :)

File: human_aware_rl/static/human_data/README.md
Comment (micahcarroll) [inline human_aware_rl/static/human_data/README.md:3]: Data collection code is proprietary? As in our psiturk experiment repos? Those should be open source too?

File: human_aware_rl/static/human_data/README.md
Comment (micahcarroll) [inline human_aware_rl/static/human_data/README.md:23]:
```suggestion
  * Data is again divided into 2019 (changed to be forward compatible with new dynamics) and 2020 experiments
```

**Review by micahcarroll — APPROVED** [review]

File: .gitmodules
Comment (nathan-miller23) [reply .gitmodules:4]: Yup! All the unit tests in the workflow will fail if it's changed back before the other merge. I'll update it once tomato_featurization is merged

File: human_aware_rl/human/temp.py
Comment (nathan-miller23) [reply human_aware_rl/human/temp.py:1]: This was committed accidentally... will remove it before merging

File: human_aware_rl/static/human_data/README.md
Comment (nathan-miller23) [reply human_aware_rl/static/human_data/README.md:3]: yeah the 'data collection code' refers to [overcooked-interactive-exp](https://github.com/micahcarroll/overcooked-interactive-exp) which is currently private. If you think it should be public we should probably add it to the HumanCompatibleAI github group, and then eventually make it part of `Overcooked-Demo`

File: human_aware_rl/static/human_data/README.md
Comment (nathan-miller23) [reply human_aware_rl/static/human_data/README.md:23]: The 2019 data (as it's stored in the repo currently) doesn't actually have the dynamics updates. The schema for the CSV, as well as the OvercookedState serialization, was forward ported, but the necessary 'interact's for soup cooking are not included

Comment (nathan-miller23) [reply human_aware_rl/static/human_data/README.md:23]: The `process_human_data.py` script includes out-of-the-box functionality for adding those additional interact actions.

My thought process was that, since we include the script to add the interacts (but don't include, or rather, don't have, a script for removing them) we should include the raw data that is missing the additional interacts. This way, users can run the script to add them should they so desire, or work with the original trajectories directly should they so choose

**Review by mesutyang97 — APPROVED** [review]
Thanks for all the great work! Let's merge this and get everything updated.

**Comment by nathan-miller23** [comment]
Great! I merged the overcooekd-ai PR and made a new PyPI release. I'll revert the submodule here and merge this as well

**Pull request merged**
